tests/legacy-cli/e2e/utils/process.ts

```diff
@@ -56,20 +59,36 @@ function _exec(options: ExecOptions, cmd: string, args: string[]): Promise<Proce
     .join(', ')
     .replace(/^(.+)$/, ' [$1]'); // Proper formatting.
 
-  console.log(colors.blue(`Running \`${cmd} ${args.map((x) => `"${x}"`).join(' ')}\`${flags}...`));
-  console.log(colors.blue(`CWD: ${cwd}`));
-
-  const spawnOptions: SpawnOptions = {
+  const spawnOptions: SpawnOptions & Required<Pick<SpawnOptions, 'env'>> = {
     cwd,
     env: { ...env, PATH: paths },
   };
 
-  if (process.platform.startsWith('win')) {
+  // Set by jobs running the e2e runner inside WSL, but want to verify CLI
+  // works well for our Windows users outside of WSL. See `.github/workflows/pr.yml`.
+  if (windowsMode !== null) {
+    // Translate command path to a native Windows path, as we execute
+    // via CMD outside WSL.
     args.unshift('/c', cmd);
-    cmd = 'cmd.exe';
-    spawnOptions['stdio'] = 'pipe';
+    cmd = windowsMode.cmdPath;
+
+    spawnOptions.stdio = 'pipe';
+    // Set WSLENV to propagate configured environment variables.
+    // E.g. NPM registry or other variables.
+    spawnOptions.env['WSLENV'] = createWslEnv(spawnOptions.env);
+
+    // Convert all absolute paths to native Windows paths that are
+    // valid in CMD outside WSL. Relative imports should still work.
+    // Absolute paths will otherwise be invalid when simulating commands
+    // for native Windows users.
+    for (let i = 0; i < args.length; i++) {
+      args[i] = interopWslPathForOutsideIfNecessary(args[i]);
+    }
   }
 
+  console.log(colors.blue(`Running \`${cmd} ${args.map((x) => `"${x}"`).join(' ')}\`${flags}...`));
+  console.log(colors.blue(`CWD: ${cwd}`));
```

Comment:
From testing?

Reply:
No, this was there before. I just fixed it to also show our "translated Windows command"